lib/mysql_connection.cpp

```diff
@@ -817,7 +823,38 @@ void MySQL_Connection::connect_start() {
 		}
 	}
 	if (parent->port) {
-		async_exit_status=mysql_real_connect_start(&ret_mysql, mysql, parent->address, userinfo->username, auth_password, userinfo->schemaname, parent->port, NULL, client_flags);
+
+		char* host_ip = NULL;
+		const std::string& res_ip = MySQL_Monitor::dns_lookup(parent->address, false);
+
+		if (!res_ip.empty()) {
+			if (connected_host_details.hostname) {
+				if (strcmp(connected_host_details.hostname, parent->address) != 0) {
+					connected_host_details.hostname = (char*)realloc(connected_host_details.hostname, strlen(parent->address) + 1);
+					strcpy(connected_host_details.hostname, parent->address);
```

Comment:
I see a mix of `realloc` + `strcpy` , or `strdup` .
Personally I find more clear to always use `strdup` , and `free` when needed.